admin/projects.go

```diff
@@ -278,19 +281,43 @@ func (s *Service) UpdateOrgDeploymentAnnotations(ctx context.Context, org *datab
 	return nil
 }
 
-// RedeployProject de-provisions and re-provisions a project's prod deployment.
-func (s *Service) RedeployProject(ctx context.Context, proj *database.Project, prevDepl *database.Deployment) (*database.Project, error) {
+// RedeployProject de-provisions and re-provisions a project's deployment.
+func (s *Service) RedeployProject(ctx context.Context, proj *database.Project, prevDepl *database.Deployment, environment string) (*database.Project, error) {
```

Comment:
It feels like passing `environment` creates some edge cases, could `prevDepl` be enough? Because then there are just two cases:
- If `prevDepl` is passed, redeploy it (and use `prevDepl.Environment` as the env)
- If `prevDepl` is not passed, "RedeployProject" means (re)deploy the primary `prod` deployment for the project.